processor/redactionprocessor/config.go

```diff
@@ -34,3 +50,16 @@ type Config struct {
 	// configuration. Possible values are `debug`, `info`, and `silent`.
 	Summary string `mapstructure:"summary"`
 }
+
+func (cfg *Config) Validate() error {
+	return cfg.validateHashFunction()
+}
+
+func (cfg *Config) validateHashFunction() error {
```

Comment:
Instead of doing this through validate, we can make `HashFunction` implement [`encoding.TextUnmarshaler`](https://pkg.go.dev/encoding#TextUnmarshaler) which is handled automatically by our configuration system